### PR TITLE
Add Railway extension

### PR DESCRIPTION
**Pull request by coffee-cup**
## Description

This extension allows you to search and navigate to your projects on [Railway][https://railway.app). Before the projects can be searched you must first login to Railway through Raycast. This process is streamlined and should only require 1 click (to verify a login code). The authentication token is stored in `${environment.supportPath}/config.json`.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/3044853/137563679-f4c650c0-bb88-4362-b1a6-3ada2819df6f.mp4

## Checklist

- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/railway/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "railway",
+  "title": "Railway",
```
Comment (thomaspaulmann) [inline extensions/railway/package.json:3]: How about naming it like this? Might look better in the store. If you ad more commands, we can rename the extension easily.

```suggestion
  "title": "Railway Project Search",
```

File: extensions/railway/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "description": "Search and navigate to your Railway projects",
+  "icon": "command-icon@light.png",
+  "author": "coffee-cup",
```
Comment (thomaspaulmann) [inline extensions/railway/package.json:6]: Nice username!

File: extensions/railway/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "railway-projects",
+      "title": "Railway Projects",
```
Comment (thomaspaulmann) [inline extensions/railway/package.json:11]: We follow the <Verb> <Noun> pattern for command names. 
```suggestion
      "title": "Search Projects",
```

File: extensions/railway/src/railway-projects.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+              title={p.name}
+              subtitle={p.description}
+              actions={
+                <ActionPanel>
+                  <ActionPanel.Section>
+                    <OpenInBrowserAction title="Project settings" url={projectUrl(p.id, "settings")} />
+                    <OpenInBrowserAction title="Project deployments" url={projectUrl(p.id, "deployments")} />
+                    <OpenInBrowserAction title="Latest deployment" url={projectUrl(p.id, "deployments?open=true")} />
+                    <OpenInBrowserAction title="Project variables" url={projectUrl(p.id, "variables")} />
+                    <OpenInBrowserAction title="Project metrics" url={projectUrl(p.id, "metrics")} />
```
Comment (thomaspaulmann) [inline extensions/railway/src/railway-projects.tsx:111]:
```suggestion
                    <OpenInBrowserAction title="Open Settings" url={projectUrl(p.id, "settings")} />
                    <OpenInBrowserAction title="Open Deployments" url={projectUrl(p.id, "deployments")} />
                    <OpenInBrowserAction title="Open Latest Deployment" url={projectUrl(p.id, "deployments?open=true")} />
                    <OpenInBrowserAction title="Open Variables" url={projectUrl(p.id, "variables")} />
                    <OpenInBrowserAction title="Open Metrics" url={projectUrl(p.id, "metrics")} />
```

File: extensions/railway/src/railway-projects.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+                  </ActionPanel.Section>
+
+                  <ActionPanel.Section>
+                    <LogoutAction token={token} />
```
Comment (thomaspaulmann) [inline extensions/railway/src/railway-projects.tsx:115]: Top that you have thought about this!

File: extensions/railway/src/railway-projects.tsx (hunk 4 above)
Comment (thomaspaulmann) [reply extensions/railway/src/railway-projects.tsx:111]: Just made all action names title case. Would it also make sense to add another section with some `CopyToClipboardAction`? Could be handy to quickly share stuff...

BTW you can also assign [keyboard shortcuts](https://developers.raycast.com/api-reference/user-interface/action-panel#actionpanel.item) to actions, e.g. `⌘` `L` could open the latest deployment.

File: extensions/railway/src/railway-projects.tsx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  }, []);
+
+  return (
+    <List isLoading={loginSession == null}>
+      {loginSession != null && (
+        <List.Item
+          title="Login to Railway"
+          actions={
+            <ActionPanel title="#1 in raycast/extensions">
+              <PushAction title="Start Railway login" target={<LoginWaiting loginSession={loginSession} />} />
```
Comment (thomaspaulmann) [inline extensions/railway/src/railway-projects.tsx:152]:
```suggestion
              <PushAction title="Start Railway Login" target={<LoginWaiting loginSession={loginSession} />} />
```

File: extensions/railway/src/railway-projects.tsx [hunk 7]
```diff
@@ -104,11 +105,37 @@ const ListProjects: React.FC<{ token: string }> = ({ token }) => {
...
+                      title="Project deployments"
+                      url={projectUrl(p.id, "deployments")}
+                    />
+                    <OpenInBrowserAction
+                      title="Latest deployment"
+                      url={projectUrl(p.id, "deployments?open=true")}
+                      shortcut={{ modifiers: ["cmd"], key: "l" }}
+                    />
+                    <OpenInBrowserAction
+                      title="Project variables"
```
Comment (pernielsentikaer) [inline extensions/railway/src/railway-projects.tsx:122]:
```suggestion
                      title="Project Variables"
```

File: extensions/railway/src/railway-projects.tsx [hunk 8]
```diff
@@ -104,11 +105,37 @@ const ListProjects: React.FC<{ token: string }> = ({ token }) => {
...
+                      url={projectUrl(p.id, "variables")}
+                      shortcut={{ modifiers: ["cmd"], key: "v" }}
+                    />
+                    <OpenInBrowserAction
+                      title="Project metrics"
```
Comment (pernielsentikaer) [inline extensions/railway/src/railway-projects.tsx:127]:
```suggestion
                      title="Project Metrics"
```

File: extensions/railway/src/railway-projects.tsx [hunk 9]
```diff
@@ -104,11 +105,37 @@ const ListProjects: React.FC<{ token: string }> = ({ token }) => {
               actions={
                 <ActionPanel>
                   <ActionPanel.Section>
-                    <OpenInBrowserAction title="Project settings" url={projectUrl(p.id, "settings")} />
-                    <OpenInBrowserAction title="Project deployments" url={projectUrl(p.id, "deployments")} />
-                    <OpenInBrowserAction title="Latest deployment" url={projectUrl(p.id, "deployments?open=true")} />
-                    <OpenInBrowserAction title="Project variables" url={projectUrl(p.id, "variables")} />
-                    <OpenInBrowserAction title="Project metrics" url={projectUrl(p.id, "metrics")} />
+                    <OpenInBrowserAction
+                      title="Project settings"
```
Comment (pernielsentikaer) [inline extensions/railway/src/railway-projects.tsx:109]:
```suggestion
                      title="Project Settings"
```

File: extensions/railway/src/railway-projects.tsx [hunk 10]
```diff
@@ -104,11 +105,37 @@ const ListProjects: React.FC<{ token: string }> = ({ token }) => {
...
+                      url={projectUrl(p.id, "settings")}
+                    />
+                    <OpenInBrowserAction
+                      title="Project deployments"
```
Comment (pernielsentikaer) [inline extensions/railway/src/railway-projects.tsx:113]:
```suggestion
                      title="Project Deployments"
```

File: extensions/railway/src/railway-projects.tsx [hunk 11]
```diff
@@ -104,11 +105,37 @@ const ListProjects: React.FC<{ token: string }> = ({ token }) => {
...
+                      url={projectUrl(p.id, "deployments")}
+                    />
+                    <OpenInBrowserAction
+                      title="Latest deployment"
```
Comment (pernielsentikaer) [inline extensions/railway/src/railway-projects.tsx:117]:
```suggestion
                      title="Latest Deployment"
```

**Pull request merged**